hca/upload/upload_area.py

```diff
@@ -98,10 +100,45 @@ def list(self, detail=False):
             for file_info in files_info:
                 yield file_info
 
-    def upload_file(self, file_path, dcp_type=None, target_filename=None, use_transfer_acceleration=True,
-                    report_progress=False):
-        file_s3_key = "%s/%s" % (self.uuid, target_filename or os.path.basename(file_path))
-        content_type = str(DcpMediaType.from_file(file_path, dcp_type))
+    def upload_files(self, file_paths, dcp_type="data", target_filename=None, use_transfer_acceleration=True,
+                     report_progress=False):
+        """
+        A function that takes in a list of file paths and other optional args for parallel file upload
+        """
+        self._setup_s3_agent_for_file_upload(file_paths=file_paths, use_transfer_acceleration=use_transfer_acceleration)
+        pool = ThreadPool()
+        if report_progress:
+            print("\nStarting upload of %s files to upload area %s" % (len(file_paths), self.uuid))
+        for file_path in file_paths:
+            pool.add_task(self._upload_file, file_path,
+                          target_filename=target_filename,
```

Comment:
Does this mean that all the `target_filename`s have to be the same for every uploaded file?

Reply:
That's a good point.  Individual filename naming doesn't with with either folder upload or parallelization.  We need to be able to turn off parallelization and prohibit folder upload if the filename option is provided.

Reply:
Marcus it sounds like this is a feature you are still using in anger?

Reply:
It is, though there are easy workarounds. I think I'm probably unusual in that I'm using the hca python library rather than the cli tool. The latter does have some safeguards that stop you from uploading multiple files and specifying a target name.